src/deployment/configuration.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import json
+import logging
+from typing import List, Optional
+from uuid import UUID
+
+from azure.cosmosdb.table.tableservice import TableService
+
+storage_client_logger = logging.getLogger("azure.cosmosdb.table.common.storageclient")
+TABLE_NAME = "InstanceConfig"
+
+logger = logging.getLogger("deploy")
+
+
+## Disable logging from storageclient. This module displays an error message
+## when a resource is not found even if the exception is raised and handled internally.
+## This happen when a table does not exist. An error message is displayed but the exception is
+## handled by the library.
+def disable_storage_client_logging() -> None:
+    if storage_client_logger:
+        storage_client_logger.disabled = True
+
+
+def enable_storage_client_logging() -> None:
+    if storage_client_logger:
+        storage_client_logger.disabled = False
+
+
+def create_if_missing(table_service: TableService) -> None:
+    try:
+        disable_storage_client_logging()
+
+        if not table_service.exists(TABLE_NAME):
+            table_service.create_table(TABLE_NAME)
+    finally:
+        enable_storage_client_logging()
+
+
+def update_allowed_aad_tenants(
+    table_service: TableService, resource_group: str, tenants: List[UUID]
+) -> None:
+    create_if_missing(table_service)

```

Comment:
The fact that we are constantly calling `create_if_missing()` suggests that maybe we should have a wrapper class that has as member a known-to-exist (via the constructor) `TableService`. It could also encapsulate a specific `resource_group`. Then we only need to ensure the existence of the resource once.